library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/Ac3Reader.java

```diff
@@ -188,18 +192,36 @@ private boolean skipToNextSync(ParsableByteArray pesBuffer) {
   private void parseHeader() {
     headerScratchBits.setPosition(0);
     Ac3Util.Ac3SyncFrameInfo frameInfo = Ac3Util.parseAc3SyncframeInfo(headerScratchBits);
-    if (format == null || frameInfo.channelCount != format.channelCount
-        || frameInfo.sampleRate != format.sampleRate
-        || frameInfo.mimeType != format.sampleMimeType) {
+    if ((format == null || frameInfo.sampleRate != format.sampleRate
+        || frameInfo.mimeType != format.sampleMimeType)
```

Comment:
Dolby's 7.1 EAC3 stream consists of I(Independent frame) and D(dependent frame).  These two frames cross appear in the stream, i.e. I,D,I,D,......
When first I frame occurs, format = null, streamType = TYPE0. So one format is created. At this time, channelCount = 6. 
Then first D frame occurs, streamType = TYPE1, channelCount = 4. So one new format is created to replace the old one.
When second I frame occurs, format != null. So no new format is created.
When second D frame occurs, maybeReconfigure = false. So no new format is created.
......
I remember channel count in each I frame (and D frame) can't be changed in AC3/EAC3 stream. But I need to double check with the original author of this piece of code. If this is not true, I think we need to modify this code a little.